python/device/device.py

```diff
@@ -119,7 +131,16 @@ def __init__(self):
         self.send_message_unacked_failure_count = 200
 
         # How many send_message exceptions do we allow before we fail the test?
-        self.send_message_exception_failure_count = 5
+        self.send_message_exception_failure_count = 1
+
+        # How often do we want the service to send test C2D messages?
+        self.receive_message_interval = 2
+
+        # How big, in bytes, do we we want the random text size in test C2D messages to be
+        self.receive_message_filler_size = 16 * 1024
+
+        # How many missing C2D messages will cause the test to fail?
+        self.receive_message_missing_message_failure_count = 1
```

Comment:
I worry we're being too aggressive about killing a test for very strict conditions. I'd rather we try to go as long as we can (no catastrophic failures), and log the issues we see along the way. Then, from the logs, we can calculate the effective run time based on our (possibly changing) criteria. Even later, if we update our criteria, older runs that we didn't kill, can still be re-evaluated.

Reply:
good point.  You've said that before and I think I'm seeing the light.  I'll adjust.